flocker/apiclient/test/test_client.py

```diff
@@ -798,6 +815,21 @@ def test_logging(self, logger):
         return d
 
     @capture_logging(None)
+    def test_persistent_connections(self, logger):
+        """
+        ``FlockerClient`` can use a pool with persistent connections when
+        specified.
```

Comment:
I _thought_ that a non-persistent pool was just like a pool with standing size of 0 that boosts to more on-demand ... effectively just a source of connections to a given server. I thought it was just misnamed pool because the implementation for persistent connections would obviously have to be a pool.

Obviously I was wrong, thanks for digging in! Removing all this nonsense as treq doesn't appear to offer a good way to do what we want to do.